KubeArmor/kvmAgent/kvmAgent.go

```diff
@@ -123,6 +127,16 @@ func (kvm *KVMAgent) ConnectToKVMService() {
 			if err == io.EOF {
 				continue
 			} else if err != nil {
+
+				if strings.Contains(string(err.Error()), errIdentityRemoved) {
```

Comment:
It is better to apply this check after the grpc connection close below. This way we can avoid repeatative grpc connection close.

Reply:
Hi @nyrahul It is not possible to use this error handling below/after the grpc connection close. GRPC connection close will be successful and the err is nil. Hence the condition is not valid. That is the reason for handling the error handling for identity-removal before.